yt/yt/server/lib/io/io_engine.h

```diff
@@ -131,44 +135,51 @@ struct IIOEngine
 
     virtual TFuture<TReadResponse> Read(
```

Comment:
Why does IOEngine know about sessions now? This is clearly not the right level of abstraction.

Reply:
> Session Id is some kind of higher-level element (blob_session, blob_chunk, location, etc.). IOEngine should not know about it.

I agree, but it already knows, and for a long time. I think it is used in other IOEngines too, so this problem is outside of this PR.

SessionId is used as a fallback when load doesn't pass a tag. In ideal scenario (when everything is tagged) it should not be used

Reply:
But specifically here I see a clear extension of the interface. Please show me where IOEngine already knows about sessions?

Reply:
it is used internally only for logging, it looks like a violation of encapsulation.